CI for metadata imports is broken

Most of the imports are broken, see https://github.com/research-software-ecosystem/content/actions/runs/12729380328

This is due to an outdated setup-python action version and/or an outdated python version:

```
2025-01-12T01:43:49.3651396Z ##[group]Run actions/setup-python@v1
2025-01-12T01:43:49.3651643Z with:
2025-01-12T01:43:49.3651818Z   python-version: 3.7
2025-01-12T01:43:49.3652022Z   architecture: x64
2025-01-12T01:43:49.3652205Z ##[endgroup]
2025-01-12T01:43:49.4102667Z ##[error]Version 3.7 with arch x64 not found
```

this breaks:
- biotools-import
- openebench-import
- bioconda-import
- biii-import
- debianmed-import

Additionally, first tests indicate that upgrading python version breaks bioconda dependencies:

```
Collecting PyYAML==6.0 (from -r ../bioconda-import/requirements.txt (line 8))
  Using cached PyYAML-6.0.tar.gz (124 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [54 lines of output]
      running egg_info
      writing lib/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/home/hmenager/utils/biotools-import/.venv312/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/hmenager/utils/biotools-import/.venv312/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/hmenager/utils/biotools-import/.venv312/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 288, in <module>
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
          return distutils.core.setup(**attrs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 186, in setup
          return run_commands(dist)
                 ^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 202, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 983, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/dist.py", line 999, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 1002, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/command/egg_info.py", line 312, in run
          self.find_sources()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/command/egg_info.py", line 320, in find_sources
          mm.run()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/command/egg_info.py", line 543, in run
          self.add_defaults()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/command/egg_info.py", line 581, in add_defaults
          sdist.add_defaults(self)
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/command/sdist.py", line 109, in add_defaults
          super().add_defaults()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/sdist.py", line 239, in add_defaults
          self._add_defaults_ext()
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/command/sdist.py", line 324, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<string>", line 204, in get_source_files
        File "/tmp/pip-build-env-w0c0vliq/overlay/lib/python3.12/site-packages/setuptools/_distutils/cmd.py", line 120, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

This is probably due python requirements which are not compatible with python 3.12

--- a/biii-import/action.yml
+++ b/biii-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: "3.12"
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

--- a/bioconda-import/action.yml
+++ b/bioconda-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: '3.12'
     - name: download bioconda recipes
       run: |
         cd /tmp

--- a/bioconda-import/requirements.txt
+++ b/bioconda-import/requirements.txt
@@ -1,9 +1,9 @@
 click==8.1.3
 Jinja2==3.1.2
-MarkupSafe==2.1.2
+MarkupSafe==3.0.2
 mypy-extensions==1.0.0
 packaging==23.0
 pathspec==0.11.0
 platformdirs==3.0.0
-PyYAML==6.0
+PyYAML==6.0.2
 tomli==2.0.1

--- a/bioconductor-import/action.yml
+++ b/bioconductor-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.10"
+        python-version: "3.12"
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

--- a/biocontainers-import/action.yml
+++ b/biocontainers-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: "3.12"
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

--- a/biotools-import/action.yml
+++ b/biotools-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: '3.12'
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

--- a/debian-med-import/action.yml
+++ b/debian-med-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: "3.12"
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

--- a/galaxytool-import/action.yml
+++ b/galaxytool-import/action.yml
@@ -15,7 +15,7 @@ runs:
     - name: Setup Python
       uses: actions/setup-python@v5
       with:
-        python-version: "3.10"
+        python-version: "3.12"
 
     - name: Install dependencies
       run: pip install -r ${{ github.action_path }}/requirements.txt

--- a/openebench-import/action.yml
+++ b/openebench-import/action.yml
@@ -11,9 +11,9 @@ runs:
   using: "composite"
   steps:
     - name: Setup Python
-      uses: actions/setup-python@v1
+      uses: actions/setup-python@v5
       with:
-        python-version: "3.7"
+        python-version: "3.12"
     - name: Install dependencies
       run:
         pip install -r ${{ github.action_path }}/requirements.txt

